src/zope/i18nmessageid/message.py

```diff
@@ -31,18 +31,37 @@ class Message(unicode):
     message id itself implicitly serves as the default text.
     """
 
-    __slots__ = ('domain', 'default', 'mapping', '_readonly')
+    __slots__ = (
+        'domain', 'default', 'mapping', '_readonly',
+        'msgid_plural', 'default_plural', 'number'
+    )
 
-    def __new__(cls, ustr, domain=None, default=None, mapping=None):
+    def __new__(cls, ustr, domain=None, default=None, mapping=None,
+                msgid_plural=None, default_plural=None, number=None):
         self = unicode.__new__(cls, ustr)
         if isinstance(ustr, self.__class__):
             domain = ustr.domain and ustr.domain[:] or domain
             default = ustr.default and ustr.default[:] or default
             mapping = ustr.mapping and ustr.mapping.copy() or mapping
+            msgid_plural = (
+                ustr.msgid_plural and ustr.msgid_plural[:] or msgid_plural)
+            default_plural = (
+                ustr.default_plural and ustr.default_plural[:]
+                or default_plural)
+            number = ustr.number is not None and ustr.number or number
```

Comment:
Can `ustr.number` be `0`?  Because this and-or trick will treat it the same as `None` and can in fact be shortened to `ustr.number or number`.

Maybe it's time to use if-else expressions instead of and-or tricks?

(Also I'm somewhat surprised that `_(_(..., default='foo'), default='bar').default` is 'foo' and not 'bar', but maybe if I took the time to read the doctests I'd see justification for it.)